checkout-ui-custom/src/utils/deleteAddress.js

```diff
@@ -0,0 +1,38 @@
+import { showAlertBox } from '../partials/Deliver/utils';
+import {
+  getAddressByName,
+  removeAddressFromDB,
+  removeAddressFromMasterData,
+  removeAddressFromOrderForm,
+} from './services';
+
+// Handle address deletion
+const handleDeleteAddress = async (addressName) => {
+  try {
+    const address = await getAddressByName(addressName);
+    // Remove address from all relevant sources
+    const results = await Promise.allSettled([
+      removeAddressFromDB(address),
+      removeAddressFromOrderForm(addressName),
+      removeAddressFromMasterData(address.id),
```

Comment:
@SegopotsoMakhutja I think we can do a try {} catch here.
No need to prevent the user from checking out if MasterData fails/